cmd/operator-migrate/kubebuilder/cmd.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubebuilder
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/operator-framework/operator-sdk/internal/util/projutil"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+// TODO: create these docs.
+const migrationDocLink = "https://operator-sdk.netlify.app/docs/migration/kubebuilder"
+
+type migrateCmd struct {
+	ptype   string
+	fromDir string
+	toDir   string
+	license string
+	owner   string
+	repo    string
+}
+
+// NewCmd returns a command that will add source code to an existing non-go operator
+func NewCmd() *cobra.Command {
+	c := &migrateCmd{}
+
+	cmd := &cobra.Command{
+		Use:   "kubebuilder",
+		Short: "migrates a legacy Operator SDK project to a Kubebuilder-style project",
+		Long: `Migrate a legacy Operator SDK project to a Kubebuilder-style project
+by scaffolding a new project using 'operator-sdk init' and adding APIs using
+'operator-sdk create api', both with current project data.
+`,
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			var ptype pluginType
+			if c.ptype != "" {
+				if ptype, err = toPluginType(c.ptype); err != nil {
+					return err
+				}
+			}
+
+			if projutil.CheckProjectRoot() != nil {
+				if c.fromDir == "" {
+					return fmt.Errorf("--from must be set if not running in a project")
+				}
+				if c.ptype == "" {
+					return fmt.Errorf("--type must be set if not running in a project")
+				}
+			} else {
+				if c.fromDir == "" {
+					c.fromDir = "."
+				}
+				if c.ptype == "" {
+					switch t := projutil.GetOperatorType(); t {
+					case projutil.OperatorTypeGo:
+						ptype = pluginTypeGo
+					case projutil.OperatorTypeUnknown:
+						log.Fatalf("Migration is not supported for project type %s", t)
+					}
+				}
+			}
+
+			projectConfig := filepath.Join(c.fromDir, "PROJECT")
+			if _, err = os.Stat(projectConfig); err == nil || os.IsExist(err) {
+				log.Fatalf("Project %s is already migrated", c.fromDir)
+			}
+
+			if err = c.runForType(ptype); err != nil && !errors.As(err, &needsHelpErr{}) {
+				log.Fatal(err)
+			}
+			return err
+		},
+	}
+
+	cmd.Flags().StringVar(&c.ptype, "type", "",
```

Comment:
I think the tool would just make sense for GO operators, so would be required to have a type here?